### ISSUE TITLE
Delete segments command

### ISSUE THREAD

**Issue by spetz** [apache/iggy#68]
Similar to the existing `DeletePartitions` command, we could introduce another kind of "administrative" request for deleting the segments. Given the specific Stream ID + Topic ID + Partition ID, the `segments_count` should remove all the segments starting from the oldest one (the lowest start offset in the log filename). The manual segments removal should act in the same way as the automated message expiration feature.

**Comment by hubcio** [comment]
I will try. 🦀

**Comment by ilikepi63** [comment]
All good if I pick this up?

**Comment by spetz** [comment]
Yes, I think it'd be nice to have the manual way of cleaning up the segments. Please take a look at how is the `DeletePartitions` implemented - starting from the command definition, and then moving to the binary and HTTP handlers, along with the appropriate tests.

**Comment by ilikepi63** [comment]
https://github.com/apache/iggy/pull/1624
